### PR TITLE
new doc: player-compatibility-testing

### PR DESCRIPTION
**Pull request by suspect15**
Submitting initial draft for DC/QA feedback and discussion

### CODE REVIEW

File: docs/guidelines/content/adding-hashes.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ description: Describes the process for adding hashes to primary achievement sets
 
 Adding hashes to a set can allow for a more enjoyable experience for players by adding translations, graphic improvements, or more regional support, however, this comes with risks that must be understood and properly mitigated. This document will explain the requirements for adding hashes to primary achievement sets.
 
+Players interested in doing compatibility testing for new hashes should reference [Player Compatibilty Testing](/guidelines/content/player-compatibility-testing)
```
Comment (wescopeland) [inline docs/guidelines/content/adding-hashes.md]:
```suggestion
Players interested in doing compatibility testing for new hashes should reference [Player Compatibility Testing](/guidelines/content/player-compatibility-testing).
```

File: docs/guidelines/content/player-compatibility-testing.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Benefits
+Because the test hashes will be linked to the set during testing, testers will keep any achievements they unlock. They will also earn beaten, completion and mastery credit as appropriate.
+
+# Request, Approval, and Testing Process
+Players interested in conducting new hash testing will contact the set maintainer via site direct message and request to conduct compatibility testing. A set maintainer is any active developer who is an author on a set. If all set authors are inactive, [QATeam](https://retroachievements.org/user/QATeam) is the set maintainer.
```
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
Players interested in conducting new hash testing should contact the set maintainer via site direct message to request to conduct compatibility testing. A set maintainer is any active developer who is an author on a set. If all set authors are inactive, [QATeam](https://retroachievements.org/user/QATeam) is the set maintainer.
```

File: docs/guidelines/content/player-compatibility-testing.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+## Restrictions and Limitations
+The following restrictions and limitations apply:
+
+- Set maintainers may deny any request at their discretion
+- Set maintainer may terminate a compatibility test at their discretion
```
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
- The set maintainer may terminate a compatibility test at their discretion
```

File: docs/guidelines/content/player-compatibility-testing.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+- Testers must use the emulator and core specified by the set maintainer to conduct all testing
+- Depending on how long a set is likely to take to master, a reasonable time limit will be determined by the set maintainer for the tester to complete the set
+- Test hashes that are not actively being tested will be unlinked and associated tests terminated
+- Only one tester will be approved per test hash
+- Set maintainer may set additional requirements at their discretion
```
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
- The set maintainer may set additional requirements at their discretion
```

File: docs/guidelines/content/player-compatibility-testing.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+### Step 4 - The Set Maintainer Links the Test Hash
+The set maintainer will link the hash and label it **COMPATIBILITY TEST - NOT FOR PLAYER USE** and inform the tester.
+
+### Step 5 - Testing the Hash
+The player will do the following during the compatibility test:
+
+- Make the compatibility testing their #1 priority on RetroAchievements by dedicating the majority of their play time to the test
+- Make a save state prior to each achievement to the maximum extent possible
+- Attempt to earn every achievement in the set that is required by the set maintainer. Sets may require full mastery/completion at the set maintainer's discretion.
```
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
- Attempt to earn every achievement in the set as required by the set maintainer. Sets may require full mastery or completion at the set maintainer's discretion.
```

File: docs/guidelines/content/player-compatibility-testing.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+- Immediately inform the set maintainer via site message when an incompatible achievement is discovered. The player will be responsible for providing the set maintainer whatever save states they request in order to modify achievement logic necessary for compatibility.
+- The set maintainer may direct additional tests after adjusting logic at their discretion. Testers are required to comply in order for the test to continue
+
+### Step 6 - Test Completion
+When the tester masters/completes the set, the test will be considered concluded. At this time, the set maintainer will make the decision whether or not to officially support the test hash. If supported, the set maintainer must submit the patch via the RAPatches forum on the Discord server or by opening a pull request on the RAPatches GitHub repository. Once the patch is added to the repository, the hash description, label, and patch URL must be updated.
```
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
When the tester masters/completes the set, the test will be considered concluded. At this time, the set maintainer will decide whether to officially support the test hash. If supported, the set maintainer must submit the patch via the RAPatches forum on the Discord server or by opening a pull request on the RAPatches GitHub repository. Once the patch is added to the repository, the hash description, label, and patch URL must be updated.
```

File: docs/guidelines/content/player-compatibility-testing.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+description: Describes the process for players to conduct compatibility testing for additional hashes to primary achievement sets on RetroAchievements.
+---
+
+# Player Compatibility Testing
+The biggest barrier to adding additional hash support to sets is the time required to thoroughly test existing achievement sets against new hashes. This document explains the process by which players may request approval to conduct hash compatibility testing.
+
+## Process Overview
+If approved for a compatibility test, the requested hash will be linked with a **COMPATIBILITY TEST - NOT FOR PLAYER USE** label. The compatibility tester will then be required to master or complete the set within a specified, reasonable time frame. The tester will make save states prior to as many achievements as possible. The tester will report issues directly to the set maintainer via site message and shall not open tickets if using the test hash. If the tester is able to validate that the set is fully compatible through mastery/completion, the hash will be renamed to its appropriate title and its patch will be added to RAPatches. The hash will be considered supported at this point.
```
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
If approved for a compatibility test, the requested hash will be linked with a **COMPATIBILITY TEST - NOT FOR PLAYER USE** label. The compatibility tester will then be required to master or complete the set within a specified, reasonable time frame. The tester should create save states before as many achievements as possible. The tester will report issues directly to the set maintainer via site message and shall not open tickets if using the test hash. If the tester is able to validate that the set is fully compatible through mastery or completion, the hash will be renamed to its appropriate title and its patch will be added to RAPatches. The hash will be considered supported at this point.
```

File: docs/guidelines/content/player-compatibility-testing.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+Players interested in conducting new hash testing will contact the set maintainer via site direct message and request to conduct compatibility testing. A set maintainer is any active developer who is an author on a set. If all set authors are inactive, [QATeam](https://retroachievements.org/user/QATeam) is the set maintainer.
+
+## Restrictions and Limitations
+The following restrictions and limitations apply:
+
+- Set maintainers may deny any request at their discretion
+- Set maintainer may terminate a compatibility test at their discretion
+- Testers must use the emulator and core specified by the set maintainer to conduct all testing
+- Depending on how long a set is likely to take to master, a reasonable time limit will be determined by the set maintainer for the tester to complete the set
+- Test hashes that are not actively being tested will be unlinked and associated tests terminated
```
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
- Test hashes that are not actively being tested will be unlinked, and the associated tests will be ended
```

File: docs/guidelines/content/player-compatibility-testing.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+The set maintainer will link the hash and label it **COMPATIBILITY TEST - NOT FOR PLAYER USE** and inform the tester.
+
+### Step 5 - Testing the Hash
+The player will do the following during the compatibility test:
+
+- Make the compatibility testing their #1 priority on RetroAchievements by dedicating the majority of their play time to the test
+- Make a save state prior to each achievement to the maximum extent possible
+- Attempt to earn every achievement in the set that is required by the set maintainer. Sets may require full mastery/completion at the set maintainer's discretion.
+- Immediately inform the set maintainer via site message when an incompatible achievement is discovered. The player will be responsible for providing the set maintainer whatever save states they request in order to modify achievement logic necessary for compatibility.
+- The set maintainer may direct additional tests after adjusting logic at their discretion. Testers are required to comply in order for the test to continue
```
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
- The set maintainer may require additional tests after updating the logic. Testers are required to comply in order for the test to continue
```

File: docs/guidelines/content/player-compatibility-testing.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+- Set maintainers may deny any request at their discretion
+- Set maintainer may terminate a compatibility test at their discretion
+- Testers must use the emulator and core specified by the set maintainer to conduct all testing
+- Depending on how long a set is likely to take to master, a reasonable time limit will be determined by the set maintainer for the tester to complete the set
+- Test hashes that are not actively being tested will be unlinked and associated tests terminated
+- Only one tester will be approved per test hash
+- Set maintainer may set additional requirements at their discretion
+
+### Step 1 - The Player's Request
+Players must include the following in their initial compatibility test request message:
```
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
Players must include the following information in their first message to request a compatibility test:
```

File: docs/guidelines/content/player-compatibility-testing.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+If a player agrees to the terms set forth by the set maintainer, they must respond and acknowledge such.
+
+### Step 4 - The Set Maintainer Links the Test Hash
+The set maintainer will link the hash and label it **COMPATIBILITY TEST - NOT FOR PLAYER USE** and inform the tester.
+
+### Step 5 - Testing the Hash
+The player will do the following during the compatibility test:
+
+- Make the compatibility testing their #1 priority on RetroAchievements by dedicating the majority of their play time to the test
+- Make a save state prior to each achievement to the maximum extent possible
```
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
- Create a save state before each achievement as much as possible
```

File: docs/guidelines/content/player-compatibility-testing.md (hunk 7 above)
Comment (wescopeland) [inline docs/guidelines/content/player-compatibility-testing.md]:
```suggestion
If approved for a compatibility test, the requested hash will be linked with a **COMPATIBILITY TEST - NOT FOR PLAYER USE** label. The compatibility tester will then be required to master or complete the set within a specified, reasonable time frame. The tester will make save states prior to as many achievements as possible. The tester will report issues directly to the set maintainer via site message and shall not open tickets if using the test hash. If the tester validates that the set is fully compatible by mastering or completing it, the hash will be renamed. Its patch will then be added to RAPatches. The hash will be considered supported at this point.
```

**Review by wescopeland — COMMENTED** [review]
Most of this feedback is not to fix typos or grammar errors.

I'm assuming most readers of this document will actually speak English as a 2nd language. This feedback primarily focuses on simplifying things like words, tense, and sentence structure so it is more likely to be understood by someone who does not speak English as their 1st language, ie:

- "mastery/completion" --> "mastery or completion"
- Replace formal phrases with simple ones using simple vocabulary
- Break up longer sentences into simple & multiple sentences
- Avoid idiomatic English expressions such as "direct additional tests" or "to the maximum extent possible".

File: docs/guidelines/content/player-compatibility-testing.md (hunk 2 above)
Comment (suspect15) [reply docs/guidelines/content/player-compatibility-testing.md]: Manually added as suggested.

File: docs/guidelines/content/player-compatibility-testing.md (hunk 3 above)
Comment (suspect15) [reply docs/guidelines/content/player-compatibility-testing.md]: Manually added as suggested along with a few other instances.

File: docs/guidelines/content/player-compatibility-testing.md (hunk 4 above)
Comment (suspect15) [reply docs/guidelines/content/player-compatibility-testing.md]: Manually added as suggested along with a few other instances.

File: docs/guidelines/content/player-compatibility-testing.md (hunk 5 above)
Comment (suspect15) [reply docs/guidelines/content/player-compatibility-testing.md]: Manually added as suggested.

File: docs/guidelines/content/player-compatibility-testing.md (hunk 6 above)
Comment (suspect15) [reply docs/guidelines/content/player-compatibility-testing.md]: Manually added as suggested.

File: docs/guidelines/content/player-compatibility-testing.md (hunk 7 above)
Comment (suspect15) [reply docs/guidelines/content/player-compatibility-testing.md]: Manually added as suggested.

File: docs/guidelines/content/player-compatibility-testing.md (hunk 9 above)
Comment (suspect15) [reply docs/guidelines/content/player-compatibility-testing.md]: Manually added as suggested.

File: docs/guidelines/content/player-compatibility-testing.md (hunk 10 above)
Comment (suspect15) [reply docs/guidelines/content/player-compatibility-testing.md]: Manually added as suggested.

File: docs/guidelines/content/player-compatibility-testing.md (hunk 11 above)
Comment (suspect15) [reply docs/guidelines/content/player-compatibility-testing.md]: Manually added as suggested.

File: docs/guidelines/content/player-compatibility-testing.md (hunk 7 above)
Comment (suspect15) [reply docs/guidelines/content/player-compatibility-testing.md]: Manually added as suggested.

**Comment by suspect15** [comment]
QA approved the process.  Language simplification made in accordance with WCopeland's feedback.  Some additional simplification was made throughout as well.  Ready for final review/merge if acceptable.

**Review by wescopeland — APPROVED** [review]

**Pull request merged**